whctools/views.py

```diff
@@ -340,19 +378,52 @@ def list_acl_members(request, acl_pk=""):
                         "portrait_url": entry.character.portrait_url(32),
                         "name": entry.character.character_name,
                         "state":new_state.name,
-                        "action": action
+                        "action": action,
+                        "reason": entry.get_reason_for_change_display()
                     }
             
             parsed_acl_history = list(parsed_acl_history.values())
 
+    mains_and_alts = {}
+    for memb in members_on_acl:
+        user_obj = get_user_from_evecharacter(memb)
+        
+        mains_and_alts.setdefault(user_obj.id, {})
+        if "main" not in mains_and_alts[user_obj.id].keys():
+            mains_and_alts[user_obj.id]["main"] = get_main_character_from_evecharacter(memb)
+        
+        mains_and_alts[user_obj.id].setdefault("alts", []).append(memb)
+        mains_and_alts[user_obj.id].setdefault("complete_alts", get_all_characters_from_user(user_obj))
 
-    context = {
-        "members": sorted([{
+    alphabetical_mains = dict(sorted(mains_and_alts.items(), key=lambda x: x[1]["main"].character_name))
+
+    for character in alphabetical_mains.values():
+        character["main"] = {
+            "name": character["main"].character_name,
+            "corp": character["main"].corporation_name,
+            "alliance": character["main"].alliance_name,
+            "portrait_url": character["main"].portrait_url(32),
+        }
+        character["alts"]  = list(sorted([{
```

Comment:
ill probably end up refactoring these 3 loops, they dont really need to be but they got the effect done for the moment